app/controllers/preprints/provider/moderators.js

```diff
@@ -81,10 +83,15 @@ export default Controller.extend(Analytics, moderatorsQueryParams.Mixin, {
     }),
 
     deleteModerator: task(function* (moderatorInstance) {
+        const removingSelf = moderatorInstance.id === this.get('currentUser.user.id');
         try {
```

Comment:
What part in this `try` is the part that might be expected to fail? It looks like if the if the destroyRecord/unload fails on removing the self, then the user will still be redirected to `/` which might be unexpected if the delete doesn't work. 

As an aside, `return`s in `try-catch-finally` do not behave how I would expect coming from python 😖

I don't think this necessarily needs to be changed but I wanted to check with you before merging.

Reply:
@alexschiller that's a good point. Can you think of a better way then just adding another variable that gets set if it makes it past the destroyRecord part without error?